src/server/index.js

```diff
@@ -14,6 +15,31 @@ export const app = next({
 });
 const nextHandler = app.getRequestHandler();
 
+// Copied from https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions
+function escapeRegExp(string) {
+    return string.replace(/[.*+\-?^${}()|[\]\\]/g, "\\$&");
+}
+
+// buildNavigationRouteRegexp builds a regular expression that can be used to detect one of the
+// navigation routes that isn't explicitly handled elsewhere.
+function buildNavigationRouteRegexp() {
+    // Build the alternation expression for each of the navigation routes.
+    let routeAlternation = "";
+    for (let k in NavigationConstants) {
+        if (k !== "LOGIN" && k !== "LOGOUT") {
+            let v = escapeRegExp(NavigationConstants[k]);
+            if (routeAlternation === "") {
+                routeAlternation = v;
+            } else {
+                routeAlternation = `${routeAlternation}|${NavigationConstants[k]}`;
+            }
+        }
+    }
+
+    // Build and compile the full regular expression.
+    return new RegExp(`^/(${routeAlternation})`);
```

Comment:
I'm not sure I follow the logic for building `routeAlternation` here.

It looks like `escapeRegExp` is only going to be called for the very first value, otherwise the value is appended as-is to `routeAlternation`?

Otherwise, if `escapeRegExp` is supposed to called on every value, then I might have built an array, and called `.join("|")` on it to make the `routeAlternation` string.

Reply:
Oops! That was an editing error. I meant to change the expression in the `else` clause as well. Thanks for catching that.